logging/src/jvmMain/kotlin/io/spine/reflect/PackageAnnotationLookup.kt

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright 2023, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.reflect
+
+import java.lang.annotation.ElementType
+import java.lang.annotation.Repeatable
+import java.lang.annotation.Target
+
+/**
+ * Locates an annotation of type [T] for the asked package,
+ * or for any of its parental packages.
+ *
+ * This lookup is similar to [AnnotatedPackages][io.spine.reflect.AnnotatedPackages].
+ *
+ * [AnnotatedPackages][io.spine.reflect.AnnotatedPackages] eagerly traverses
+ * all loaded packages during the instance creation. It looks for ones
+ * that are annotated with the given annotation type, remembers them,
+ * and then allows retrieving of an annotation for the asked package.
+ *
+ * But as more classes are loaded by the classloader, more new packages appear.
+ * As a result, data within the collection become insufficient. An instance
```

Comment:
"... data within the collection become insufficient..."

This portion required rewriting. First of all, "data" is "it", not "they". Second of all, "insufficient" for what?